adapters/repos/db/shard_read.go

```diff
@@ -172,19 +173,45 @@ func (s *Shard) objectSearch(ctx context.Context, limit int,
 
 		bm25Config := s.index.getInvertedIndexConfig().BM25
 
-		searcher := inverted.NewBM25Searcher(bm25Config, s.store,
+		bm25searcher := inverted.NewBM25Searcher(bm25Config, s.store,
 			s.index.getSchema.GetSchemaSkipAuth(), s.invertedRowCache,
 			s.propertyIndices, s.index.classSearcher, s.deletedDocIDs, s.propLengths,
 			s.index.logger, s.versioner.Version())
+
+		var bm25objs []*storobj.Object
+		var bm25count []float32
+		var err error
+		var objs helpers.AllowList
+		var filterDocIds helpers.AllowList
+
+		if filters != nil {
+			objs, err = inverted.NewSearcher(s.store, s.index.getSchema.GetSchemaSkipAuth(),
+				s.invertedRowCache, s.propertyIndices, s.index.classSearcher,
+				s.deletedDocIDs, s.index.stopwords, s.versioner.Version()).
+				DocIDs(ctx, filters, additional, s.index.Config.ClassName)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			filterDocIds = objs
+		}
 		if keywordRanking != nil && keywordRanking.Type == "bm25" {
```

Comment:
not new code bity I just noticed that keywordRanking is used in both branches of this if. So the `keywordRanking != nil` does not seem correct to me?

Reply:
It's to prevent a segfault when reading keywordRanking.Type

Reply:
But if it is nil we go to the else branch, where keywordRanking is also used?

Reply:
Yes, because it will also go to the second branch when keyword ranking exists and Type != "bm25" (i.e. other searches).

Reply:
In addition, writing "keywordranking"  in the call to Objects shows the reader that Objects expects a keywordRanking struct, even in the circumstances where it is nil

Reply:
So in the second branch, keywordRanking can be nil OR a another search type, which is the same condition that the parent function deals with

Reply:
alright